### PR TITLE
FIX: colorbar pad for `ImageGrid`

### LINKED ISSUE #28343: [Bug]: incosistent colorbar pad for `ImageGrid` with `cbar_mode="single"`
### Bug summary

`ImageGrid` with `cbar_mode="single"` adds the `axes_pad` between the axes and the colorbar if the colorbar is to the left or at the bottom. That seems inconsistent and unnecessary as the space can be controlled with `cbar_pad`.

### Code for reproduction

```Python
import matplotlib.pyplot as plt
from mpl_toolkits.axes_grid1 import ImageGrid


fig = plt.figure(figsize=(6, 6))

subfigs = fig.subfigures(2, 2, wspace=0.07, hspace=0.1, linewidth=1)
figs = subfigs.flatten()

locations = ["left", "bottom", "right", "top"]

for fig, location in zip(figs, locations):

    fig.set_edgecolor("r")

    axgr = ImageGrid(
        fig,
        111,
        nrows_ncols=(1, 1),
        axes_pad=1. / 2.54,
        cbar_mode="single",
        cbar_location=location,
        cbar_pad=0. / 2.54,
    )
    ax = axgr.axes_all[0]
    ax.text(0.5, 0.5, f"{location=}", transform=ax.transAxes, va="center", ha="center")
    ax.set(xticks=[], yticks=[])
    
    cbax = axgr.cbar_axes[0]
    cbax.set(xticks=[], yticks=[])

    plt.savefig("cbar_mode_single.png")
```

### Actual outcome

cbar_mode_single

### Expected outcome

I expect all colorbars to hug the axes.

### Additional information

- As mentioned this only happens for `cbar_mode="single"` and `cbar_location` `"left"` or `"bottom"`
- This behavior seems to have been around since forever.
- There is the `cbar_pad` param which can be used to control the distance between axes and colorbar
- Note that both `cbar_pad` and `axes_pad` are added between the axes and the colorbar
- I am happy to open to open a PR, but don't see a way to deprecate this gracefully

---

- the space is added in these lines respectively:
   https://github.com/matplotlib/matplotlib/blob/479bd7a0cad2b77ef522c1dd6d65080a4da62b4f/lib/mpl_toolkits/axes_grid1/axes_grid.py#L442-L443
   https://github.com/matplotlib/matplotlib/blob/479bd7a0cad2b77ef522c1dd6d65080a4da62b4f/lib/mpl_toolkits/axes_grid1/axes_grid.py#L471-L472
- This is tested in
https://github.com/matplotlib/matplotlib/blob/479bd7a0cad2b77ef522c1dd6d65080a4da62b4f/lib/mpl_toolkits/axes_grid1/tests/test_axes_grid1.py#L420
- and the image: https://github.com/matplotlib/matplotlib/blob/479bd7a0cad2b77ef522c1dd6d65080a4da62b4f/lib/mpl_toolkits/axes_grid1/tests/baseline_images/test_axes_grid1/image_grid_single_bottom_label_mode_1.png

### Operating system

_No response_

### Matplotlib Version

3.8.4

### Matplotlib Backend

_No response_

### Python version

_No response_

### Jupyter version

_No response_

### Installation

conda

### PR DESCRIPTION
**Pull request by mathause**
## PR summary

This PR removes the inconsistent pad between the axes and the colorbar of `ImageGrid` with `cbar_mode="single"` and therefore closes #28343. Instead of changing the test image I updated the image creation. I am not entirely sure I made the "changelog" entries correctly for mpl - let me know if any changes are required.

## PR checklist

- [ ] "closes #0000" is in the body of the PR description to [link the related issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)
- [ ] new and changed code is [tested](https://matplotlib.org/devdocs/devel/testing.html)
- [ ] *Plotting related* features are demonstrated in an [example](https://matplotlib.org/devdocs/devel/document.html#write-examples-and-tutorials)
- [ ] *New Features* and *API Changes* are noted with a [directive and release note](https://matplotlib.org/devdocs/devel/api_changes.html#announce-changes-deprecations-and-new-features)
- [ ] Documentation complies with [general](https://matplotlib.org/devdocs/devel/document.html#write-rest-pages) and [docstring](https://matplotlib.org/devdocs/devel/document.html#write-docstrings) guidelines

### CODE REVIEW

File: doc/users/next_whats_new/mpl_toolkit_colorbar_pad.rst [hunk 1]
```diff
@@ -0,0 +1,6 @@
+Fix padding of single colorbar for ``ImageGrid``
+------------------------------------------------
+
+``ImageGrid`` with ``cbar_mode="single"`` no longer adds the ``axes_pad`` between the
+axes and the colorbar for thr ``cbar_location`` left and bottom. Add required space
+using `cbar_pad` instead.
```
Comment (timhoffm) [inline doc/users/next_whats_new/mpl_toolkit_colorbar_pad.rst:6]:
```suggestion
axes and the colorbar for ``cbar_location`` "left" and "bottom". If desired, add additional spacing
unsing ``cbar_pad``.
```

File: lib/mpl_toolkits/axes_grid1/axes_grid.py [hunk 2]
```diff
@@ -358,6 +358,11 @@ def __init__(self, fig,
         cbar_location : {"left", "right", "bottom", "top"}, default: "right"
         cbar_pad : float, default: None
             Padding between the image axes and the colorbar axes.
+
+            .. versionchanged:: 3.10
+                `cbar_mode="single"` no longer adds the `axes_pad` between the axes and
+                the colorbar if the `cbar_location` is `"left"` or `"bottom"`
```
Comment (timhoffm) [inline lib/mpl_toolkits/axes_grid1/axes_grid.py:364]:
```suggestion
                ``cbar_mode="single"`` no longer adds *axes_pad* between the axes
                and the colorbar if the *cbar_location* is "left" or "bottom".
```
ReST formatting and some conventions from https://matplotlib.org/devdocs/devel/document.html#formatting-conventions.

**Review by timhoffm — COMMENTED** [review]
This should fix the doc build failure.

File: doc/users/next_whats_new/mpl_toolkit_colorbar_pad.rst [hunk 3]
```diff
@@ -0,0 +1,6 @@
+Fix padding of single colorbar for ``ImageGrid``
+------------------------------------------------
+
+``ImageGrid`` with ``cbar_mode="single"`` no longer adds the ``axes_pad`` between the
+axes and the colorbar for ``cbar_location`` "left" and "bottom". If desired, add additional spacing
+unsing ``cbar_pad``.
```
Comment (mathause) [inline doc/users/next_whats_new/mpl_toolkit_colorbar_pad.rst:6]:
```suggestion
using ``cbar_pad``.
```

**Review by timhoffm — APPROVED** [review]

**Review by ksunden — APPROVED** [review]

**Pull request merged**